opm/io/eclipse/EclFile.hpp

```diff
@@ -118,4 +124,68 @@ class EclFile
 
 }} // namespace Opm::EclIO
 
+
+namespace Opm {
+    namespace EclIO {
+    
+    class EclFileLGR : public EclFile {
+    public:
+        EclFileLGR(const std::string& filename, bool preload = false)
+            : EclFile(filename, preload)
+        {
+            initialize_lgr_properties();
+            classify_lgr();
+        };
+    
+        EclFileLGR(const std::string& filename, Formatted fmt, bool preload = false)
+            : EclFile(filename, fmt, preload)
+        {
+            initialize_lgr_properties();
+            classify_lgr();
+        };
+        
+        template <typename T>
+        const std::vector<T>& get(int arrIndex)
+        {
+            return EclFile::get<T>(arrIndex);
+        };
+      
+        template <typename T>
+        const std::vector<T>& get(const std::string& name){
+            return EclFile::get<T>(name);
+        };
+        template <typename T>
+        const std::vector<T>& get(const std::string& name, const std::string& lgr_name = "GLOBAL")
+        {
+            const auto tag = array_name_to_num[name];
+            const int lgr_num = lgr_labels[lgr_name];
```

Comment:
These statements insert elements into `array_name_to_num` and `lgr_labels` if `name` or `lgr_name` do not exist in the maps already (side effect of [`map<>::operator[]()`](https://en.cppreference.com/w/cpp/container/map/operator_at)).  Is that what you want?  If not, you should use `map<>::find()` instead of `map<>::operator[]()`.